src/core/Ekubo.sol

```diff
@@ -350,11 +354,12 @@ abstract contract Ekubo is SettlerAbstract {
 
                 // TODO: verify that the vault enforces that the settled sell amount cannot be
                 // positive
-                state.sell.amount -= uint256(settledSellAmount.unsafeNeg());
+                state.sell.amount -= uint256(settledSellAmount);
+
                 // If `state.buy.amount()` overflows an `int128`, we'll get a revert inside the
                 // vault later. We cannot overflow a `uint256`.
                 unchecked {
-                    state.buy.amount += settledBuyAmount.asCredit(state.buy.token);
+                    state.buy.amount += settledBuyAmount.unsafeNeg().asCredit(state.buy.token);
```

Comment:
I think you should use `asDebt` instead. Of course, add a comment that the sign convention here is backwards from UniV4/BalV3/PancakeInfinity